### ISSUE TITLE
False positives for 932260 with POST parameters starting with axel

### ISSUE THREAD

**Issue by hydrapolic** [coreruleset/coreruleset#3564]
### Description

A POST parameter contains axel and is blocked because it's found in:

```
rules/unix-shell.data:bin/axel
```

Seems to me it's similar to https://github.com/coreruleset/coreruleset/issues/3490.

### Logs

```
---RCgaAsui---B--
POST / HTTP/1.0
Content-Type: application/x-www-form-urlencoded
```

```
---RCgaAsui---C--
&param1=1&email=axeluser%40email.com&param2=2
```

```
---RCgaAsui---H--
ModSecurity: Warning. Matched "Operator `Rx' with parameter `[0-9]\s*\'\s*[0-9]' against variable `MATCHED_VAR' (Value: `axeluser@email.com' ) [file "/usr/share/modsecurity-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"] [line "478"] [id "932260"] [rev ""] [msg "Remote Command Execution: Direct Unix Command Execution"] [data "Matched Data: axel found within MATCHED_VAR: axeluser@email.com"] [severity "2"] [ver "OWASP_CRS/4.0.0-rc2"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-shell"] [tag "platform-unix"] [tag "attack-rce"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/152/248/88"] [tag "PCI/6.5.2"] ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:BLOCKING_INBOUND_ANOMALY_SCORE' (Value: `5' ) [file "/usr/share/modsecurity-crs/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "176"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "0"] [ver "OWASP_CRS/4.0.0-rc2"] [maturity "0"] [accuracy "0"] [tag "anomaly-evaluation"] [ref ""]
```

### Your Environment

* CRS version (e.g., v3.3.4): 4.0.0-rc2
* Paranoia level setting (e.g. PL1) : PL1
* ModSecurity version (e.g., 2.9.6): 3.0.12
* Web Server and version or cloud provider / CDN (e.g., Apache httpd 2.4.54): Nginx 1.25.3
* Operating System and version: Linux Gentoo

### Confirmation

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by dune73** [comment]
Thanks for reporting and sorry for the inconvenience. I think this has been reported - and fixed before. Could you test with the latest nightly?

**Comment by hydrapolic** [comment]
> Thanks for reporting and sorry for the inconvenience. I think this has been reported - and fixed before. Could you test with the latest nightly?

By looking at the dev branch, it seems like it was changed on other places. It's still in unix-shell.data:  https://github.com/coreruleset/coreruleset/blob/v4.0/dev/rules/unix-shell.data#L63

**Comment by hydrapolic** [comment]
It's strange that `unix-shell.data` contains `bin/axel`, however the email only starts with `axel`, there is no `bin` there.

**Comment by dune73** [comment]
The ways of the RCE rules are dark and complex...

But I confirm this FP is gone in nightly. The following call only throws an alert in PL4.

```
$ curl "localhost/?param1=1&email=axeluser%40email.com&param2=2"
```

**Comment by hydrapolic** [comment]
Just tested with latest nightly, you're right it's gone.

I've missed the fact that `crs-toolchain regex update -a` was run and it updated the regexp that contained the fix. It cannot be simply grepped.

**Comment by hydrapolic** [comment]
Thanks for the help @dune73

**Comment by dune73** [comment]
You're most welcome.

In fact you got me nervous. We are planning to release CRSv4 later today and this FP would be really bad in the release...

**Comment by hydrapolic** [comment]
Sorry for that @dune73.

However, I've deployed CRSv4 on a few projects and overall it's doing fine.

**Comment by dune73** [comment]
No worries. And pleased to hear that.
